### PR TITLE
Fix RIPEMD160 support

### LINKED ISSUE #78: Ripemd160 throws when used
``` js
/lib/sync-browser.js:51
  h.copy(this.opad, this.blocksize)
    ^

TypeError: Cannot read property 'copy' of undefined
```

I don't think this path is under test at all.

### PR DESCRIPTION
**Pull request by timonegk**
The RIPEMD160 module is not correctly used in the current implementation. This PR fixes that (and closes #78).

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
can we get a test for this so we don't revert this again ?

**Comment by timonegk** [comment]
So I added tests to the `fixtures.json` file. However, one case (the one with the 'abandon' key) fails in the async variant because the password is derived to `undefined`. It works when I try it manually though. Do you have any idea why this happens?

File: lib/sync-browser.js [hunk 1]
```diff
@@ -57,7 +57,12 @@ function getDigest (alg) {
     return sha(alg).update(data).digest()
   }
 
-  if (alg === 'rmd160' || alg === 'ripemd160') return rmd160
+  if (alg === 'rmd160' || alg === 'ripemd160') {
+    var rmd160Func = function (data) {
+      return new RIPEMD160().update(data).digest()
+    }
+    return rmd160Func
```
Comment (dcousens) [inline lib/sync-browser.js]: shouldn't be inlined

**Comment by dcousens** [comment]
ACK,   but why aren't we using `create-hash` again?

**Comment by calvinmetcalf** [comment]
crypto-pouch doesn't use RIPEMD so @kgathuru this is probably unrelated

@dcousens we are not using create-hash because that library has overhead due to creating stream objects and other stuff to deal with the node api which would slow down this library

**Pull request merged**